core/test/integration/api/api_invites_spec.js

```diff
@@ -411,6 +467,60 @@ describe('Invites API', function () {
                     done(new Error('Author should not be able to add an Author'));
                 }).catch(checkForErrorType('NoPermissionError', done));
             });
+
+            it('CANNOT invite a Contributor', function (done) {
+                InvitesAPI.add({
+                    invites: [
+                        {
+                            email: 'test@example.com',
+                            role_id: testUtils.roles.ids.contributor
+                        }
+                    ]
+                }, context.author).then(function () {
+                    done(new Error('Author should not be able to add a Contributor'));
+                }).catch(checkForErrorType('NoPermissionError', done));
+            });
+        });
+
+        describe('Contributor', function () {
+            it('CANNOT invite an Owner', function (done) {
+                InvitesAPI.add({
+                    invites: [
+                        {
+                            email: 'test@example.com',
+                            role_id: testUtils.roles.ids.owner
+                        }
+                    ]
+                }, context.contributor).then(function () {
+                    done(new Error('Contributor should not be able to add an owner'));
+                }).catch(checkForErrorType('NoPermissionError', done));
+            });
+
+            it('CANNOT invite an Author', function (done) {
+                InvitesAPI.add({
+                    invites: [
+                        {
+                            email: 'test@example.com',
+                            role_id: testUtils.roles.ids.author
+                        }
+                    ]
+                }, context.contributor).then(function () {
+                    done(new Error('Contributor should not be able to add an Author'));
+                }).catch(checkForErrorType('NoPermissionError', done));
+            });
+
+            it('CANNOT invite a Contributor', function (done) {
+                InvitesAPI.add({
+                    invites: [
+                        {
+                            email: 'test@example.com',
+                            role_id: testUtils.roles.ids.contributor
+                        }
+                    ]
+                }, context.contributor).then(function () {
+                    done(new Error('Contributor should not be able to add a Contributor'));
+                }).catch(checkForErrorType('NoPermissionError', done));
+            });
```

Comment:
Also most of these tests can be transformed into unit tests?

Reply:
We agreed on extending this test:

> /ghost/core/test/integration/migration_spec.js

The approach is:

- the permission service is 100% unit tested
- the permissible functions should be 100% united tested for the contributor role
- we use one integration test to ensure that the correct permission relations are inserted for the contributor role
- keep some routing tests if the API has custom permission code (e.g. invites API, this needs refactoring - we could maybe use the invite permissible function and unit test)